pkg/vm/vm.go

```diff
@@ -1055,6 +1055,11 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 				panic("zero modulus") // https://docs.microsoft.com/en-us/dotnet/api/system.numerics.biginteger.modpow?view=net-6.0#exceptions
 			}
 			res.Exp(base, exponent, modulus)
+
+			if base.Sign() < 0 && exponent.Bit(0) == 1 {
+				absModulus := new(big.Int).Abs(modulus)
+				res.Sub(res, absModulus)
```

Comment:
If the result is `0`, then `res.Sub(res, absModulus)` should not be applied, isn't it? I need you to build a testcase where `base ^ exp` is a multiple of `mod` (for negative `base` and even `exp`) to check what happens in case of zero result.

Reply:
true. missed this case